tools/wptserve/wptserve/response.py

```diff
@@ -451,6 +451,11 @@ def write_headers(self, headers, status_code, status_message=None, stream_id=Non
         secondary_headers = []  # Non ':' prefixed headers are to be added afterwards
 
         for header, value in headers:
+            # h2_header by default is unicode in PY3
```

Comment:
This goes deeper than a constant happening to be unicode, or we could just change it to bytes. I haven't been able to chase the rabbit-hole fully, however.

If one makes `h2_handlers` a list of bytes and makes the obvious changes, it then seems like `H2HandlerCopy` needs to decode its `self.path` variable which is used in the `Request` class and seems like its expected to be unicode:

```

        self.request_path = request_handler.path
        self.url_base = "/"

        if self.request_path.startswith(scheme + "://"):   <--- needs to be unicode then
```

I'm not *opposed* to the changes here, but I think it highlights more than ever that we need to put some time aside ahead of us to properly convert wptserve to be sensible and consistent about bytes-vs-strings. This may become easier once we are in Py3-only land and don't need to worry about Py2, not sure.

Reply:
I can't find `h2_handlers`?

`request_path` is indeed a native string, one of the few remaining in fact (#23362).

Reply:
Argh, my typo. `h2_headers` in `tools/wptserve/wptserve/constants.py`